### PR TITLE
[AWIBOF-7111] modifiy reset segment condition. 

### PR DESCRIPTION
**Pull request by dhihm**
code changes:
modifiy reset segment condition while resetting segments info. Occupied stripe count can be 0 if there was no flushed stripe.

### CODE REVIEW

File: src/allocator/context_manager/segment_ctx/segment_ctx.cpp [hunk 1]
```diff
@@ -617,9 +617,8 @@ SegmentCtx::ResetSegmentsStates(void)
         }
         else
         {
-            POS_TRACE_ERROR(EID(ALLOCATOR_FILE_ERROR), "segment id {}, validCount {}, occupiedStripeCount {}",
+            POS_TRACE_DEBUG(EID(ALLOCATOR_RESET_SEGMENTS), "segment id {}, validCount {}, occupiedStripeCount {}",
```
Comment (huijeong-kim) [inline src/allocator/context_manager/segment_ctx/segment_ctx.cpp:620]: NVRAM으로 가는 조건을 0 <= occupiedStripeCount 로 하는건 어떤가요 ?.?

Comment (dhihm) [reply src/allocator/context_manager/segment_ctx/segment_ctx.cpp:620]: 넵 수정하겠습니다~!

Comment (dhihm) [reply src/allocator/context_manager/segment_ctx/segment_ctx.cpp:620]: 기존 assert는 다시 살려 놓는게 좋겠군요.

File: doc/troubleshooting/events.md [hunk 2]
```diff
@@ -1,1109 +1,1110 @@
...
-|   4061   |                     MFS_QUEUE_POP_FAILED                    |  None  |                                             None                                            |                                                         None                                                        |                                        None                                       |
-|   4062   |                    MFS_QUEUE_PUSH_FAILED                    |  None  |                                             None                                            |                                                         None                                                        |                                        None                                       |
-|   4063   |                  MFS_MAX_FILE_SIZE_NOT_SET                  |  None  |                                             None                                            |                                                         None                                                        |                                        None
```
Comment (dhihm) [inline doc/troubleshooting/events.md:1]: Clion 으로 편집 했더니 자동으로 indent 해버려서 다시 수정해서 올렸습니다. 3213번에 ALLOCATOR_RESET_SEGMENTS 추가 했습니다.

**Review by huijeong-kim — DISMISSED** [review]
고생많으셨습니다!

File: doc/troubleshooting/events.md [hunk 3]
```diff
@@ -1,1109 +1,1110 @@
-|    Id    |                             Name                            |Severity|                                           Message                                           |                                                        Cause                                                        |                                      Solution                                     |
```
Comment (yyu-nim) [inline doc/troubleshooting/events.md:1]: 이 파일은 delete commit 이 들어가는게 맞을까요?? 혹시 의도된 것인지 확인 부탁드리겠습니다

File: src/allocator/context_manager/segment_ctx/segment_ctx.cpp [hunk 4]
```diff
@@ -611,13 +611,13 @@ SegmentCtx::ResetSegmentsStates(void)
         {
             segmentInfos[segId].SetState(SegmentState::SSD);
         }
-        else if ((0 <= validCount) && (1 <= occupiedStripeCount))
+        else if ((0 <= validCount) && (0 <= occupiedStripeCount))
```
Comment (yyu-nim) [inline src/allocator/context_manager/segment_ctx/segment_ctx.cpp:614]: 오... 이런 boundary condition 이 있었군요 혹시 UT 로 의도를 나타내고, 미래에 저희가 모두 없을 때 누군가가 이걸 1로 다시 바꿀때 test failure로 warning을 주는 것이 좋지 않을까요??? 예를 들면
```
TEST(SegmentCtx, ResetSegmentsState_testIfSegmentStateBecomesNVRAMWhenOccupiedStripeCountIsZeroDuringLogReplay) {

}
```
대략 이런 느낌으로...

File: doc/troubleshooting/events.md (hunk 3 above)
Comment (dhihm) [reply doc/troubleshooting/events.md:1]: 네 indent가 변경되어서, 원복+ 3213번 추가했습니다.

File: src/allocator/context_manager/segment_ctx/segment_ctx.cpp [hunk 5]
```diff
@@ -611,13 +611,13 @@ SegmentCtx::ResetSegmentsStates(void)
...
         {
             segmentInfos[segId].SetState(SegmentState::NVRAM);
         }
         else
         {
-            POS_TRACE_ERROR(EID(ALLOCATOR_FILE_ERROR), "segment id {}, validCount {}, occupiedStripeCount {}",
+            POS_TRACE_INFO(EID(ALLOCATOR_FILE_ERROR), "segment id {}, validCount {}, occupiedStripeCount {}",
```
Comment (yyu-nim) [inline src/allocator/context_manager/segment_ctx/segment_ctx.cpp:620]: ERROR -> INFO 로 log level 낮춘것이 의도된 것일지 확인부탁드릴께요

**Review by yyu-nim — COMMENTED** [review]
코멘트 하나 남겨두었는데 그것만 확인하고 approve 하도록 하겠습니다. 분석&수고 많으셨습니다!

File: src/allocator/context_manager/segment_ctx/segment_ctx.cpp (hunk 4 above)
Comment (dhihm) [reply src/allocator/context_manager/segment_ctx/segment_ctx.cpp:614]: 네 추가해 두겠습니다. ^^

File: src/allocator/context_manager/segment_ctx/segment_ctx.cpp (hunk 5 above)
Comment (dhihm) [reply src/allocator/context_manager/segment_ctx/segment_ctx.cpp:620]: 아래가 assert라서 큰 의미가 없을 것 같긴 한데, 기존대로 원복해 두겠습니다. ^^

File: test/unit-tests/allocator/context_manager/segment_ctx/segment_ctx_test.cpp [hunk 6]
```diff
@@ -851,4 +851,45 @@ TEST(SegmentCtx, SetRebuildCompleted_testIfSegmentIsRemovedFromTheList)
...
+
+    segmentCtx.ResetSegmentsStates();
+
+    EXPECT_EQ(SegmentState::NVRAM, segInfos[0].GetState());
+    EXPECT_EQ(SegmentState::NVRAM, segInfos[1].GetState());
+    EXPECT_EQ(SegmentState::SSD, segInfos[2].GetState());
+    EXPECT_EQ(SegmentState::FREE, segInfos[3].GetState());
+
+    delete[] segInfos;
+}
```
Comment (yyu-nim) [inline test/unit-tests/allocator/context_manager/segment_ctx/segment_ctx_test.cpp:894]: 모범적인 UT case 짜주셨네요 ㅎㅎ 감사합니다!

**Review by yyu-nim — APPROVED** [review]
수고 많으셨습니다!!

**Review by MunseopLim — APPROVED** [review]
출근을 안해서 task가 기억이 안나는데..ㅎㅎ
이게 count가 맞지 않고 else에서 assert 걸렸던거였죠?
고생하셨습니다~
